### ISSUE TITLE
Incorrect second-order sphere traversal from slashlink in content

### ISSUE THREAD

**Issue by cdata** [subconsciousnetwork/subconscious#564]
This issue is a follow-up after investigating https://github.com/subconsciousnetwork/noosphere/issues/320

TL;DR I think that whatever is responding to slashlinks being tapped in a note is trying to traverse from the wrong origin sphere (in the case of a second-order traversal).

When performing a second-order traversal (to a peer of my peer), I noticed that I can perform the traversal via the address book of my peer, but if I try to do the same traversal by clicking a slashlink in my peer’s note it fails. See below attached video for an example.

In the example, I first traverse to `@gordon`, and then to `@gordon`'s `try-again-fail-better` slug, and from there I attempt a second-order traversal by clicking the link `@cdata/all-your-base` (which fails). I then demonstrate that I can navigate to `@cdata.gordon` via `@gordon`'s address book.

In diagnosing this issue, I enabled full logging of Noosphere code paths and watched the output in XCode. The logs can tell us which spheres we are traversing from and to whenever a traversal happens. The logs clearly show that when the failed second-order traversal happens, the sphere we are traversing from is not `@gordon`'s, but `@cdata`'s. That is to say: we are effectively trying to go to `@cdata.cdata` (a petname sequence which does not exist) when we actually mean to go to `@cdata.gordon`.

This can be confusing because if I have `@ben` in my address book, and there is also a `@ben.gordon`, such a second-order traversal will appear to work even though it is erroneously traversing from my sphere, not `@gordon`'s.

An example of the relevant logs follows. Note that in both traversals the "origin" is my sphere (`did:key:z6MkjiAUCg157324ekAE2W33YBuWxvVo7fW2pSuUMZS6XqcZ`), even though in the traversal to `cdata` the origin __should__ be `@gordon.cdata` (`did:key:z6MkseshcgDTj2X6vWmqurQ9rJxqhr7jL3HDz2CWzgZ7hGPp`):

```
2023-04-29 00:49:35.649509-0700 Subconscious[3014:1593844] [SubtextTextViewRepresentable] updateUIView
2023-04-29 00:49:35.650095-0700 Subconscious[3014:1593844] [detail] [action] autosave
2023-04-29 00:49:36.170057-0700 Subconscious[3014:1593844] [SubtextTextViewRepresentable] updateUIView
2023-04-29 00:49:37.749978-0700 Subconscious[3014:1593844] [notebook] [action] pushDetail(Subconscious.MemoDetailDescription.viewer(Subconscious.MemoViewerDetailDescription(address: public::@gordon/try-again-fail-better)))
2023-04-29 00:49:37.777834-0700 Subconscious[3014:1593844] [SubtextTextViewRepresentable] updateUIView
  2023-04-29T07:49:37.785285Z DEBUG noosphere_sphere::context: Petname assigned to IdentityIpld { did: Did("did:key:z6MkseshcgDTj2X6vWmqurQ9rJxqhr7jL3HDz2CWzgZ7hGPp"), link_record: Some(Link { cid: Cid(bafk2bzacebrvqqipwa46qvngocpzvsakgzfiifu2fxrwph6k5lyygs6jqkcvc), linked_type: PhantomData<noosphere_core::data::address::LinkRecord> }) }
    at rust/noosphere-sphere/src/context.rs:147
    in noosphere_sphere::context::traverse_by_petname with petname: "gordon", origin: "did:key:z6MkjiAUCg157324ekAE2W33YBuWxvVo7fW2pSuUMZS6XqcZ"

  2023-04-29T07:49:37.785411Z DEBUG noosphere_sphere::context: Resolved version is bafy2bzaceaolca73ykllhx2a3xvlssuyobf7ffoy4w3m44yp4nsscji3cbqma
    at rust/noosphere-sphere/src/context.rs:164
    in noosphere_sphere::context::traverse_by_petname with petname: "gordon", origin: "did:key:z6MkjiAUCg157324ekAE2W33YBuWxvVo7fW2pSuUMZS6XqcZ"

  2023-04-29T07:49:37.785428Z DEBUG noosphere_sphere::context: Local version: bafy2bzaceaolca73ykllhx2a3xvlssuyobf7ffoy4w3m44yp4nsscji3cbqma
    at rust/noosphere-sphere/src/context.rs:170
    in noosphere_sphere::context::traverse_by_petname with petname: "gordon", origin: "did:key:z6MkjiAUCg157324ekAE2W33YBuWxvVo7fW2pSuUMZS6XqcZ"

  2023-04-29T07:49:37.785455Z DEBUG noosphere_sphere::context: Checking to see if we can get the sphere body...
    at rust/noosphere-sphere/src/context.rs:195
    in noosphere_sphere::context::traverse_by_petname with petname: "gordon", origin: "did:key:z6MkjiAUCg157324ekAE2W33YBuWxvVo7fW2pSuUMZS6XqcZ"

  2023-04-29T07:49:37.785508Z  INFO noosphere::ffi::context: Reading sphere did:key:z6MkseshcgDTj2X6vWmqurQ9rJxqhr7jL3HDz2CWzgZ7hGPp slug try-again-fail-better...
    at rust/noosphere/src/ffi/context.rs:297

  2023-04-29T07:49:37.785791Z DEBUG noosphere_sphere::content::decoder: Unpacking block bafy2bzaceawg7cdjh4yp5vtkzhlpma2l5xoeqoep6tctajsw2j2wzzypoffmk...
    at rust/noosphere-sphere/src/content/decoder.rs:18

2023-04-29 00:49:37.785978-0700 Subconscious[3014:1593844] [MemoViewerDetail] [action] appear(Subconscious.MemoViewerDetailDescription(address: public::@gordon/try-again-fail-better))
2023-04-29 00:49:37.789690-0700 Subconscious[3014:1593844] [SubtextTextViewRepresentable] updateUIView
2023-04-29 00:49:37.817886-0700 Subconscious[3014:1593844] [SubtextTextViewRepresentable] updateUIView
2023-04-29 00:49:37.818201-0700 Subconscious[3014:1593844] [MemoViewerDetail] [action] setDetail(...)
ForEach<Array<Block>, Block, Text>: the ID  occurs multiple times within the collection, this will give undefined results!
2023-04-29 00:49:41.104299-0700 Subconscious[3014:1593844] [notebook] [action] findAndPushDetail(slashlink: @cdata/_profile_, fallback: "_profile_")
2023-04-29 00:49:41.136093-0700 Subconscious[3014:1593844] [SubtextTextViewRepresentable] updateUIView
  2023-04-29T07:49:41.142372Z  WARN noosphere_sphere::context: "cdata" is not assigned to an identity
    at rust/noosphere-sphere/src/context.rs:142
    in noosphere_sphere::context::traverse_by_petname with petname: "cdata", origin: "did:key:z6MkjiAUCg157324ekAE2W33YBuWxvVo7fW2pSuUMZS6XqcZ"
```

https://user-images.githubusercontent.com/240083/235290213-c4387e5c-fb0f-48d2-bd60-d622631fe739.MP4
